test/cpp/interop/stress_test.cc

```diff
@@ -0,0 +1,221 @@
+}
+
+// Input: Test case string "<testcase_name:weight>,<testcase_name:weight>...."
+// Output:
+//   - Whether parsing was successful (return value)
+//   - Vector of (test_type_enum, weight) pairs
+bool ParseTestCasesString(const string& test_cases,
+                          vector<pair<TestCaseType, int>>& tests) {
+  bool is_success = true;
+
+  vector<string> tokens;
+  ParseCommaDelimitedString(test_cases, tokens);
+
+  for (auto it = tokens.begin(); it != tokens.end(); it++) {
+    // Token is in the form <test_name>:<test_weight>
+    size_t colon_pos = it->find(':');
+    if (colon_pos == string::npos) {
+      gpr_log(GPR_ERROR, "Error in parsing test case string: %s", it->c_str());
+      is_success = false;
+      break;
+    }
+
+    string test_name = it->substr(0, colon_pos);
+    int weight = std::stoi(it->substr(colon_pos + 1));
+    TestCaseType test_case = get_test_type_from_name(test_name);
+
+    if (test_case == UNKNOWN_TEST) {
+      gpr_log(GPR_ERROR, "Unknown test case: %s", test_name.c_str());
+      is_success = false;
+      break;
+    }
+
+    tests.emplace_back(std::make_pair(test_case, weight));
+  }
+
+  return is_success;
+}
+
+// For debugging purposes
+void LogParameterInfo(const vector<string>& addresses,
+                      const vector<pair<TestCaseType, int>>& tests) {
+  gpr_log(GPR_INFO, "server_addresses: %s", FLAGS_server_addresses.c_str());
+  gpr_log(GPR_INFO, "test_cases : %s", FLAGS_test_cases.c_str());
+  gpr_log(GPR_INFO, "sleep_duration_ms: %d", FLAGS_sleep_duration_ms);
+  gpr_log(GPR_INFO, "test_duration_secs: %d", FLAGS_test_duration_secs);
+
+  int num = 0;
+  for (auto it = addresses.begin(); it != addresses.end(); it++) {
+    gpr_log(GPR_INFO, "%d:%s", ++num, it->c_str());
+  }
+
+  num = 0;
+  for (auto it = tests.begin(); it != tests.end(); it++) {
+    TestCaseType test_case = it->first;
+    int weight = it->second;
+    gpr_log(GPR_INFO, "%d. TestCaseType: %d, Weight: %d", ++num, test_case,
+            weight);
+  }
+}
+
+int main(int argc, char** argv) {
+  grpc::testing::InitTest(&argc, &argv, true);
+  gpr_log(GPR_INFO, "starting test..");
+
+  srand(time(NULL));
+
+  // Parse the server addresses
+  vector<string> server_addresses;
+  ParseCommaDelimitedString(FLAGS_server_addresses, server_addresses);
+
+  // Parse test cases and weights
+  if (FLAGS_test_cases.length() == 0) {
+    gpr_log(GPR_INFO, "Test cases string is empty");
+    return 1;
+  }
+
+  vector<pair<TestCaseType, int>> tests;
+  if (!ParseTestCasesString(FLAGS_test_cases, tests)) {
+    gpr_log(GPR_ERROR, "Error in parsing test cases string %s ",
+            FLAGS_test_cases.c_str());
+    return 1;
+  }
+
+  LogParameterInfo(server_addresses, tests);
+
+  TestSelector test_selector(tests);
+
+  vector<thread> test_threads;
+  int thread_idx = 0;
+  for (auto it = server_addresses.begin(); it != server_addresses.end(); it++) {
+    StressTestInteropClient* client = new StressTestInteropClient(
```

Comment:
the std::thread() constructor expects a function pointer. Which is why I passed it as 
```thread(&StressTestInteropClient::MainLoop, client)``` . I don't know to do this if client is of type unique_ptr... (i can't do client.get() obviously)